src/node/handler/PadMessageHandler.js

```diff
@@ -181,10 +181,11 @@ exports.handleMessage = async function(client, message)
   var env = process.env.NODE_ENV || 'development';
 
   if (env === 'production') {
+    const clientIPAddress = (settings.trustProxy && client.handshake.headers['x-forwarded-for']) ? client.handshake.headers['x-forwarded-for'] : client.handshake.address;
```

Comment:
```suggestion
    const clientIPAddress = remoteAddress[client.id];
```

Reply:
Perfect. I saw this, but did not realize it's set in the connection setup phase and is available to later socket.io messages.